test/js-native-api/test_constructor/test.js

```diff
@@ -48,3 +48,14 @@ assert.throws(() => { test_object.readonlyAccessor2 = 3; },
 // to the instance
 assert.strictEqual(TestConstructor.staticReadonlyAccessor1, 10);
 assert.strictEqual(test_object.staticReadonlyAccessor1, undefined);
+
+// Verify that passing NULL to napi_define_class() results in the correct
+// error.
+assert.deepStrictEqual(TestConstructor.TestDefineClass(), {
```

Comment:
Could the assertion be made in `test/js-native-api/test_constructor/test_constructor.c:TestDefineClass` directly and make a simple not throw assertion(comment maybe) here?

Reply:
@legendecas we've already landed tests in a similar vein here: https://github.com/nodejs/node/pull/27628/

Since this PR also introduces `NULL`-guards which would otherwise cause segfaults I'd be OK with landing as is and maybe doing a follow-up PR.